paddle/fluid/operators/distributed/request_handler_impl.cc

```diff
@@ -113,11 +118,15 @@ bool RequestPrefetchHandler::Handle(const std::string& varname,
                                     const std::string& out_var_name) {
   VLOG(4) << "RequestPrefetchHandler " << varname;
 
-  auto var_desc = program_->Block(0).FindVar(out_var_name);
-  InitializeVariable(*outvar, var_desc->GetType());
-  executor_->RunPreparedContext(
-      (*prefetch_var_name_to_prepared_ctx_)[varname].get(), scope);
-
+  // All execution should use thread in the thread pool to do.
+  std::future<void> ft =
+      framework::Async([this, &scope, &varname, &out_var_name, &outvar]() {
+        auto var_desc = program_->Block(0).FindVar(out_var_name);
+        InitializeVariable(*outvar, var_desc->GetType());
+        executor_->RunPreparedContext(
+            (*prefetch_var_name_to_prepared_ctx_)[varname].get(), scope);
+      });
+  ft.wait();
```

Comment:
I still suspect this pattern really works. What if `RequestPrefetchHandler::Handle` is put in a threadpool? Then it's thread in thread?

Reply:
If `RequestPrefetchHandler::Handle` is in thread_pool, then this change is not needed.